src/views/ReservasView.java

```diff
@@ -296,7 +339,29 @@ public void propertyChange(PropertyChangeEvent evt) {
 			public void mouseClicked(MouseEvent e) {
 				if (ReservasView.txtFechaEntrada.getDate() != null && ReservasView.txtFechaSalida.getDate() != null) {		
 					RegistroHuesped registro = new RegistroHuesped();
-					registro.setVisible(true);
+					
+					Date entryDatePrevious = txtFechaEntrada.getDate();
+					Date departureDatePrevious = txtFechaSalida.getDate();
+
+					
+					if (entryDatePrevious != null && departureDatePrevious != null) {
```

Comment:
This if clause is redundant, if you check currently line 340 we have this same condition.